addon/-private/system/references/belongs-to.js

```diff
@@ -63,32 +63,33 @@ BelongsToReference.prototype.push = function(objectOrPromise) {
   });
 };
 
-BelongsToReference.prototype.value = function() {
-  var inverseRecord = this.belongsToRelationship.inverseRecord;
+BelongsToReference.prototype.isLoaded = function() {
+  let inverseRecord = this.belongsToRelationship.inverseRecord;
+
+  return inverseRecord && inverseRecord.isLoaded();
+};
 
-  if (inverseRecord && inverseRecord.record) {
-    return inverseRecord.record;
+BelongsToReference.prototype.value = function() {
+  if (this.isLoaded()) {
+    return this.belongsToRelationship.inverseRecord.getRecord();
   }
 
   return null;
 };
 
 BelongsToReference.prototype.load = function() {
-  if (this.remoteType() === "id") {
-    return this.belongsToRelationship.getRecord();
-  }
+  let record = this.value();
```

Comment:
yes, we do. `inverseRecord` is actually an instance of `internalModel`. `getRecord` is the correct way of getting, well a record, from an `internalModel`. `getRecord` on the `BelongsToRelationship` is method to fetch a record (it deals internally with links and stuff).